recipe/recipe.go

```diff
@@ -0,0 +1,109 @@
+package recipe
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/ONSdigital/dp-api-clients-go/clientlog"
+	"github.com/ONSdigital/dp-api-clients-go/headers"
+	healthcheck "github.com/ONSdigital/dp-api-clients-go/health"
+	health "github.com/ONSdigital/dp-healthcheck/healthcheck"
+	"github.com/ONSdigital/log.go/log"
+)
+
+const service = "recipe-api"
+
+// Client is a recpie api client which can be used to make requests to the server
+type Client struct {
+	hcCli *healthcheck.Client
+}
+
+// New creates a new instance of Client with a given recipe api url
+func New(recipeAPIURL string) *Client {
+	return &Client{
+		healthcheck.NewClient(service, recipeAPIURL),
+	}
+}
+
+// NewWithHealthClient creates a new instance of Client,
+// reusing the URL and Clienter from the provided health check client.
+func NewWithHealthClient(hcCli *healthcheck.Client) *Client {
+	return &Client{
+		healthcheck.NewClientWithClienter(service, hcCli.URL, hcCli.Client),
+	}
+}
+
+// Checker calls recipe api health endpoint and returns a check object to the caller.
+func (c *Client) Checker(ctx context.Context, check *health.CheckState) error {
+	return c.hcCli.Checker(ctx, check)
+}
+
+// CloseResponseBody closes the response body and logs an error if unsuccessful
+func CloseResponseBody(ctx context.Context, resp *http.Response) {
+	if resp.Body == nil {
+		return
+	}
+	if err := resp.Body.Close(); err != nil {
+		log.Event(ctx, "error closing http response body", log.ERROR, log.Error(err))
+	}
+}
+
+// GetRecipe from an ID
+func (c *Client) GetRecipe(ctx context.Context, userAuthToken, serviceAuthToken, recipeID string) (*Recipe, error) {
+	uri := fmt.Sprintf("%s/recipes/%s", c.hcCli.URL, recipeID)
+	clientlog.Do(ctx, "retrieving recipe", service, uri)
```

Comment:
There was a mention in the guild meeting yesterday that perhaps we shouldn't be logging at all in any of our libraries (I removed the logging I had because of this) and let the services handle the logging.. this sounded like a good way to me, not sure what you think? It does look that logging package was specifically designed for client though..

Reply:
Yes, I added it for consistency with other services. I can remove it, and maybe at some point, we remove the logging from all other packages in the library.